client/src/main.rs

```diff
@@ -449,6 +449,47 @@ fn perform_trusted_operation(matches: &ArgMatches<'_>, top: &TrustedOperation) -
     }
 }
 
+/// Returns the next valid index (aka nonce) for given account.
+fn get_nonce_direct(matches: &ArgMatches<'_>, account: &AccountId) -> Vec<u8> {
```

Comment:
why is this here and not in stf/cli.rs?
the nonce belongs to the "trusted" space, so it shouldn't be here. If it were here, it should be the layer1 nonce that's returned
sidechain nonce should have its `TrustedGetter`

Reply:
If we implement the get_nonce function as a TrustedGetter, it needs to go through the top pool first, delaying the submission of a trusted call. If that is no problem, then changing the get nonce should not be a problem.

The function is called on the stf/cli.rs side. It is implemnted on the main.rs side because of the arguments and functions it needs. I imitated the implementation of "perform_trusted_operation".

Reply:
we need to clarify this. you can only query the sidechain nonce by loading the state which ist costly. so it should be done through the TOP pool like everything else. The alternative is that the client keeps track of the nonce itself. If in doubt, query!

Reply:
I still believe this is in the wrong place